go/teams/teams.go

```diff
@@ -850,3 +874,35 @@ func LoadTeamPlusApplicationKeys(ctx context.Context, g *libkb.GlobalContext, id
 	}
 	return team.ExportToTeamPlusApplicationKeys(ctx, keybase1.Time(0), application)
 }
+
+func (t *Team) loadAllTransitiveSubteams(ctx context.Context) ([]*Team, error) {
+	subteams := []*Team{}
+	for subteamID := range t.Data.Chain.SubteamLog {
+		// Load each subteam...
+		subteam, err := Load(ctx, t.G(), keybase1.LoadTeamArg{
+			ID:          subteamID,
+			NeedAdmin:   true,
+			ForceRepoll: true,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Force loading the key manager.
+		// TODO: Should this be the default, so that we don't need to do it here?
+		_, err = subteam.SharedSecret(ctx)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I would not be opposed to `keyManager` being set up in `NewTeam` as long as it cannot error (which I think is the case, it cannot error). Then as you said we could get rid of all these side-effecting `SharedSecret` calls.